source/client/output_formatter_impl.cc

```diff
@@ -138,5 +144,123 @@ DottedStringOutputFormatterImpl::formatProto(const nighthawk::client::Output& ou
   return ss.str();
 }
 
+const nighthawk::client::Result&
+FortioOutputFormatterImpl::getGlobalResult(const nighthawk::client::Output& output) const {
+  for (const auto& nh_result : output.results()) {
+    if (nh_result.name() == "global") {
+      return nh_result;
+    }
+  }
+
+  throw NighthawkException("Nighthawk output was malformed, contains no 'global' results.");
```

Comment:
Perhaps consider covering this throw and the ones below in tests?
As a suggestion, there are the python integration tests which might be useful for that, you can perform an end-to-end test with bad inputs, this would also ensure the throw are handled as intended:
https://github.com/envoyproxy/nighthawk/blob/master/test/integration/test_integration_basics.py

Reply:
I think it's pretty hard to simulate all 3 errors with invalid inputs to the fortio client. I think most invalid inputs will just be caught by the first `getGlobalResult` function (?).

I'll look into adding unit tests instead.

Reply:
Done!

Reply:
Thanks, looks great. 
Double checking; one gotcha is that the some consumers don't currently expect the output formatters to throw [1]. I feel that should be addressed outside of this PR, I created https://github.com/envoyproxy/nighthawk/issues/184 for tracking.